### ISSUE TITLE
False positive for SVG fragment identifiers in color-hex-length

### ISSUE THREAD

**Issue by bobbor** [stylelint/stylelint#2806]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

when using svg fragment identifiers in an background url, stylelint would suggest long/short color-variations on this identifer

> Which rule, if any, is this issue related to?

`color-hex-length`

> What CSS is needed to reproduce this issue?
```css
b {
  background: url(my.svg#view);
}
```

> What stylelint configuration is needed to reproduce this issue?
```json
{
  "rules": {
    "color-hex-length": "long"
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint  *./src/**/*.pcss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

No warnings to be flagged. Fragment identifiers should be ignored

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
Expected "#view" to be "#vviieeww"   color-hex-length
```

**Comment by bobbor** [comment]
maybe it should also be noted, that `#view` (of course) is not a color.
the regexp's in the `color` rules could be adjusted to `[0-9A-Fa-f]` to not give wrong feedback and/or a new rule could be introduced which lints invalid-hex-colors

**Comment by ntwb** [comment]
Fixed via https://github.com/stylelint/stylelint/commit/4412f8365e5757cbe1fca9d17edc5259a1a993ae
